src/main/java/io/vertx/spi/cluster/hazelcast/impl/serializable/VertxDataSerializableFactory.java

```
@@ -0,0 +1,19 @@
+package io.vertx.spi.cluster.hazelcast.impl.serializable;
+
+import com.hazelcast.nio.serialization.DataSerializableFactory;
+import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
+
+public class VertxDataSerializableFactory implements DataSerializableFactory {
+	@Override
+	public IdentifiedDataSerializable create(int id) {
+		switch (id) {
+		case 0:
+			return new BufferDataSerializable();
+		case 1:
+			return new JsonDataSerializable();
+		default:
```

be explicit and use 2 for JsonArray. default should throw an exception